include/mouse.h

```diff
@@ -21,20 +22,88 @@
 
 #include "dosbox.h"
 
-void Mouse_ShowCursor(void);
-void Mouse_HideCursor(void);
+// ***************************************************************************
+// Notifications from external subsystems - all should go via these methods
+// ***************************************************************************
 
-bool Mouse_SetPS2State(bool use);
+void Mouse_EventMoved(int32_t x_rel, int32_t y_rel, int32_t x_abs, int32_t y_abs, bool is_captured);
+void Mouse_EventPressed(uint8_t idx);
+void Mouse_EventReleased(uint8_t idx);
+void Mouse_EventWheel(int32_t w_rel);
+
+void Mouse_SetSensitivity(int32_t sensitivity_x, int32_t sensitivity_y);
+void Mouse_NewScreenParams(uint16_t clip_x, uint16_t clip_y, uint16_t res_x, uint16_t res_y,
+                           bool fullscreen, int32_t x_abs, int32_t y_abs);
+void Mouse_NotifyMovedVMW();
+
+// ***************************************************************************
+// Common structures, please only update via functions above
+// ***************************************************************************
+
+typedef struct {
+
+    float  sensitivity_x;    // sensitivity, might depend on the GUI/GFX 
+    float  sensitivity_y;    // for scaling all relative mouse movements
+
+} MouseInfoConfig;
+
+typedef struct {
+
+    bool     fullscreen;
+    uint16_t res_x;          // resolution to which guest image is scaled, excluding black borders
+    uint16_t res_y;
+    uint16_t clip_x;         // clipping = size of black border (one side)
+    uint16_t clip_y;         // clipping value - size of black border (one side)
+
+} MouseInfoVideo;
+
+extern MouseInfoConfig mouse_config;
+extern MouseInfoVideo  mouse_video;
+extern bool            mouse_vmware; // true = VMware driver took over the mouse
 
+// ***************************************************************************
+// Serial mouse
+// ***************************************************************************
+
+class CSerialMouse;
+
+void MouseSER_RegisterListener(CSerialMouse *listener);
+void MouseSER_UnRegisterListener(CSerialMouse *listener);
+
+// - needs relative movements
+// - understands up to 3 buttons
+// - needs index of button which changed state
+
+void MouseSER_NotifyMoved(int32_t x_rel, int32_t y_rel);
+void MouseSER_NotifyPressed(uint8_t buttons_12S, uint8_t idx);
+void MouseSER_NotifyReleased(uint8_t buttons_12S, uint8_t idx);
+void MouseSER_NotifyWheel(int32_t w_rel);
```

Comment:
`MouseSER` is ambiguous, and the choice of casing is funny. Could you call it `MouseSerial` instead?

Reply:
Renamed all `MouseSER_` to `MOUSESERIAL_`